src/Package/Impl/Sql/Publish/SProcGenerator.cs

```diff
@@ -0,0 +1,186 @@
+            sprocTemplate = sprocTemplate.Replace(RCodeTemplate, Environment.NewLine + rCode);
+
+            var sqlQuery = GetSqlFileContent(rFilePath.ToQueryFilePath()).Trim();
+            return sprocTemplate.Replace(InputQueryTemplate, sqlQuery);
+        }
+
+        private string FillSprocTableTemplage(string rFilePath, string sprocName, string codeTableName) {
+            var sprocTemplateFile = rFilePath.ToSProcFilePath();
+            var sprocTemplate = GetSqlFileContent(sprocTemplateFile);
+
+            var format =
+@"BEGIN
+DECLARE @RCodeQuery NVARCHAR(max);
+DECLARE @RCode NVARCHAR(max);
+DECLARE @ParmDefinition NVARCHAR(max);
+
+SET @RCodeQuery = 'SELECT @RCodeOUT = RCode FROM {0} WHERE SProcName = ''{1}''';
+SET @ParmDefinition = N'@RCodeOUT NVARCHAR(max) OUTPUT';
+
+EXEC sp_executesql @RCodeQuery, @ParmDefinition, @RCodeOUT=@RCode OUTPUT;
+SELECT @RCode;
+";
+            var declarations = string.Format(CultureInfo.InvariantCulture, format, codeTableName, sprocName);
+            sprocTemplate = sprocTemplate.Replace("BEGIN", declarations);
+            sprocTemplate = sprocTemplate.Replace("N'_RCODE_'", "@RCode");
+
+            var sqlQuery = GetSqlFileContent(rFilePath.ToQueryFilePath()).Trim();
+            return sprocTemplate.Replace(InputQueryTemplate, sqlQuery);
+        }
+
+        private void CreateStoredProcedures(SqlSProcPublishSettings settings, EnvDTE.Project targetProject, string targetFolder, EnvDTE.ProjectItem targetProjectItem) {
+            foreach (var rFilePath in settings.Files) {
+                var sprocName = settings.SProcNames[rFilePath];
+                if (!string.IsNullOrEmpty(sprocName)) {
+
+                    string template;
+                    if (settings.CodePlacement == RCodePlacement.Inline) {
+                        template = FillSprocInlineTemplate(rFilePath, sprocName);
+                    } else {
+                        template = FillSprocTableTemplage(rFilePath, sprocName, settings.TableName);
+                    }
+                    if (!string.IsNullOrEmpty(template)) {
+                        var sprocFile = Path.ChangeExtension(Path.Combine(targetFolder, sprocName), ".sql");
+                        _fs.WriteAllText(sprocFile, template);
+                        targetProjectItem.ProjectItems.AddFromFile(sprocFile);
+                    }
+                }
+            }
+        }
+
+        private string GetRFileContent(string filePath) {
+            if (_fs.FileExists(filePath)) {
+                return _fs.ReadAllText(filePath).Replace("'", "''");
+            }
+            return string.Empty;
+        }
+
+        private string GetSqlFileContent(string filePath) {
+            if (_fs.FileExists(filePath)) {
+                var sb = new StringBuilder();
+                foreach (var line in _fs.FileReadAllLines(filePath)) {
+                    if (!line.TrimStart().StartsWithOrdinal("--")) {
```

Comment:
Mostly to strip template comments so it appears more compact, but it really doesn't matter, we can keep them